0.18.1: pytest fails 128 units with `setuptools` 75.x

Looks like test suite fails with `setuptools` 75.x:
<details>
<summary>Here is pytest summary output:</summary>

```console
=========================== short test summary info ============================
SKIPPED [1] tests/test_platform.py:120: Requires Windows
SKIPPED [24] tests/test_setup.py:582: unsupported configuration: python package fully generated by CMake does *NOT* work. At least __init__.py should be in the project source tree
SKIPPED [1] tests/test_skbuild.py:84: NMake Makefiles generator is not available on Linux
SKIPPED [3] tests/test_skbuild.py:130: Requires Windows
SKIPPED [1] tests/test_skbuild.py:154: Requires Windows
FAILED tests/test_broken_project.py::test_cmakelists_with_fatalerror_fails - ...
FAILED tests/test_broken_project.py::test_cmakelists_with_syntaxerror_fails
FAILED tests/test_broken_project.py::test_hello_with_compileerror_fails - Att...
FAILED tests/test_broken_project.py::test_invalid_cmake[CalledProcessError]
FAILED tests/test_broken_project.py::test_invalid_cmake[OSError] - AttributeE...
FAILED tests/test_broken_project.py::test_first_invalid_generator - Attribute...
FAILED tests/test_broken_project.py::test_invalid_generator - AttributeError:...
FAILED tests/test_cmake_target.py::test_cmake_target_build - AttributeError: ...
FAILED tests/test_cmakelists_not_in_top_level_dir.py::test_build - AttributeE...
FAILED tests/test_cmakelists_not_in_top_level_dir.py::test_cmake_source_dir[invalid-True]
FAILED tests/test_cmakelists_not_in_top_level_dir.py::test_cmake_source_dir[-False]
FAILED tests/test_cmakelists_not_in_top_level_dir.py::test_cmake_source_dir[.-False]
FAILED tests/test_cmakelists_not_in_top_level_dir.py::test_hello_sdist - Attr...
FAILED tests/test_command_line.py::test_help - AttributeError: module 'distut...
FAILED tests/test_command_line.py::test_help_commands - AttributeError: modul...
FAILED tests/test_command_line.py::test_metadata_display - AttributeError: mo...
FAILED tests/test_command_line.py::test_no_command - AttributeError: module '...
FAILED tests/test_command_line.py::test_invalid_command - AttributeError: mod...
FAILED tests/test_command_line.py::test_too_many_separators - AttributeError:...
FAILED tests/test_command_line.py::test_cmake_initial_cache_as_global_option
FAILED tests/test_command_line.py::test_cmake_executable_arg - AttributeError...
FAILED tests/test_command_line.py::test_hide_listing[True-sdist] - AttributeE...
FAILED tests/test_command_line.py::test_hide_listing[True-bdist_wheel] - Attr...
FAILED tests/test_command_line.py::test_hide_listing[False-sdist] - Attribute...
FAILED tests/test_command_line.py::test_hide_listing[False-bdist_wheel] - Att...
FAILED tests/test_command_line.py::test_run_cmake_arg - AttributeError: modul...
FAILED tests/test_command_line.py::test_skip_cmake_arg - AttributeError: modu...
FAILED tests/test_cython_flags.py::test_hello_cython_builds - AttributeError:...
FAILED tests/test_filter_manifest.py::test_bdist_wheel_command - AttributeErr...
FAILED tests/test_hello_cpp.py::test_hello_builds - AttributeError: module 'd...
FAILED tests/test_hello_cpp.py::test_hello_sdist - AttributeError: module 'di...
FAILED tests/test_hello_cpp.py::test_hello_wheel - AttributeError: module 'di...
FAILED tests/test_hello_cpp.py::test_hello_clean[with-dry-run] - AttributeErr...
FAILED tests/test_hello_cpp.py::test_hello_clean[without-dry-run] - Attribute...
FAILED tests/test_hello_cpp.py::test_hello_cleans - AttributeError: module 'd...
FAILED tests/test_hello_cython.py::test_hello_cython_builds - AttributeError:...
FAILED tests/test_hello_cython.py::test_hello_cython_sdist - AttributeError: ...
FAILED tests/test_hello_cython.py::test_hello_cython_wheel - AttributeError: ...
FAILED tests/test_hello_fortran.py::test_hello_fortran_build - AttributeError...
FAILED tests/test_hello_fortran.py::test_hello_fortran_sdist - AttributeError...
FAILED tests/test_hello_fortran.py::test_hello_fortran_wheel - AttributeError...
FAILED tests/test_hello_pure.py::test_hello_pure_builds - AttributeError: mod...
FAILED tests/test_hello_pure.py::test_hello_pure_sdist - AttributeError: modu...
FAILED tests/test_hello_pure.py::test_hello_pure_wheel - AttributeError: modu...
FAILED tests/test_hello_pure.py::test_hello_clean - AttributeError: module 'd...
FAILED tests/test_include_exclude_data.py::test_include_exclude_data - Attrib...
FAILED tests/test_include_exclude_data.py::test_include_exclude_data_with_base
FAILED tests/test_issue284_build_ext_inplace.py::test_build_ext_inplace_command
FAILED tests/test_issue335_support_cmake_source_dir.py::test_bdist_wheel_command
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[None-cli_setup_args0-keyword_cmake_args0-cli_cmake_args0-10.9]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[10.7-cli_setup_args1-keyword_cmake_args1-cli_cmake_args1-10.7]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[10.7-cli_setup_args2-keyword_cmake_args2-cli_cmake_args2-10.9]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[None-cli_setup_args3-keyword_cmake_args3-cli_cmake_args3-10.6]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[None-cli_setup_args4-keyword_cmake_args4-cli_cmake_args4-10.7]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[None-cli_setup_args5-keyword_cmake_args5-cli_cmake_args5-10.7]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[None-cli_setup_args6-keyword_cmake_args6-cli_cmake_args6-10.7]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[None-cli_setup_args7-keyword_cmake_args7-cli_cmake_args7-10.8]
FAILED tests/test_issue342_cmake_osx_args_in_setup.py::test_cmake_args_keyword_osx_default[None-cli_setup_args8-keyword_cmake_args8-cli_cmake_args8-10.8]
FAILED tests/test_issue352_isolated_environment_support.py::test_isolated_env_trigger_reconfigure
FAILED tests/test_issue668_symbol_visibility.py::test_symbol_visibility[ON]
FAILED tests/test_issue668_symbol_visibility.py::test_symbol_visibility[OFF]
FAILED tests/test_issue707_tested_packages.py::test_install_command - Attribu...
FAILED tests/test_manifest_in.py::test_manifest_in_wheel - AttributeError: mo...
FAILED tests/test_outside_project_root.py::test_outside_project_root_fails[None]
FAILED tests/test_outside_project_root.py::test_outside_project_root_fails[-DINSTALL_FILE:BOOL=1]
FAILED tests/test_outside_project_root.py::test_outside_project_root_fails[-DINSTALL_PROJECT:BOOL=1]
FAILED tests/test_setup.py::test_cmake_args_keyword[cmake_args0] - AttributeE...
FAILED tests/test_setup.py::test_cmake_args_keyword[cmake_args1] - AttributeE...
FAILED tests/test_setup.py::test_cmake_install_dir_keyword[None-True-str] - A...
FAILED tests/test_setup.py::test_cmake_install_dir_keyword[-True-str] - Attri...
FAILED tests/test_setup.py::test_cmake_install_dir_keyword[/tmp/scikit-build-True-SKBuildError]
FAILED tests/test_setup.py::test_cmake_install_dir_keyword[banana-False-str]
FAILED tests/test_setup.py::test_cmake_with_sdist_keyword[True] - AttributeEr...
FAILED tests/test_setup.py::test_cmake_with_sdist_keyword[False] - AttributeE...
FAILED tests/test_setup.py::test_cmake_minimum_required_version_keyword - Att...
FAILED tests/test_setup.py::test_setup_requires_keyword_include_cmake - Attri...
FAILED tests/test_setup.py::test_script_keyword[pure] - AttributeError: modul...
FAILED tests/test_setup.py::test_script_keyword[skbuild] - AttributeError: mo...
FAILED tests/test_setup.py::test_py_modules_keyword[pure] - AttributeError: m...
FAILED tests/test_setup.py::test_py_modules_keyword[skbuild] - AttributeError...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-0-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-0-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-0-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-1-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-1-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-0-1-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-0-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-0-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-0-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-1-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-1-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-0-1-1-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-0-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-0-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-0-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-1-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-1-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-0-1-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-0-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-0-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-0-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-1-0-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-1-1-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[0-1-1-1-1-1] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-0-0-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-0-0-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-0-1-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-0-1-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-1-0-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-1-0-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-1-1-0-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_setup_inputs[1-1-1-1-0-0] - AttributeError: ...
FAILED tests/test_setup.py::test_cmake_install_into_pure_package[0] - Attribu...
FAILED tests/test_setup.py::test_cmake_install_into_pure_package[1] - Attribu...
FAILED tests/test_setup.py::test_zip_safe_default[None] - AttributeError: mod...
FAILED tests/test_setup.py::test_zip_safe_default[False] - AttributeError: mo...
FAILED tests/test_setup.py::test_zip_safe_default[True] - AttributeError: mod...
FAILED tests/test_skbuild.py::test_generator[Unix Makefiles-make] - Attribute...
FAILED tests/test_skbuild.py::test_invalid_generator[generator_args0] - Attri...
FAILED tests/test_skbuild.py::test_invalid_generator[generator_args1] - Attri...
FAILED tests/test_skbuild_variable.py::test_skbuild_variable_builds - Attribu...
FAILED tests/test_skbuild_variable.py::test_skbuild_variable_sdist - Attribut...
FAILED tests/test_skbuild_variable.py::test_skbuild_variable_wheel - Attribut...
===== 131 failed, 74 passed, 30 skipped, 8 deselected in 96.65s (0:01:36) ======
```
</details>
And full pytest log in attachment: 
https://github.com/user-attachments/files/17193656/python-scikit-build.FAIL.txt

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -38,7 +38,7 @@ jobs:
       with:
         fetch-depth: 0
 
-    - uses: yezz123/setup-uv@v4
+    - uses: astral-sh/setup-uv@v4
 
     - name: Set environment for available GHA MSVCs
       if: matrix.runs-on == 'windows-2019'
@@ -65,18 +65,13 @@ jobs:
     - name: Setup nox
       uses: wntrblm/nox@2024.04.15
       with:
-        python-versions: "3.7,3.8,3.9,3.10,3.11,3.12,3.13"
+        python-versions: "3.8,3.9,3.10,3.11,3.12,3.13"
 
     # We check all Python's on Linux, because it's fast.
     # We check minimum Python and maximum Python on all OS's.
 
-    - name: Test on 🐍 3.7
-      run: >
-        nox --default-venv-backend=${{ startsWith(matrix.runs-on, 'windows') && 'virtualenv' || 'uv'}}
-        -s tests-3.7 -- --cov --cov-report=xml --cov-report=term --durations=20
     - name: Test on 🐍 3.8
-      if: runner.os == 'Linux'
-      run: nox -s tests-3.8 -- --cov --cov-report=xml --cov-report=term --cov-append --durations=20
+      run: nox -s tests-3.8 -- --cov --cov-report=xml --cov-report=term --durations=20
     - name: Test on 🐍 3.9
       if: runner.os == 'Linux'
       run: nox -s tests-3.9 -- --cov --cov-report=xml --cov-report=term --cov-append --durations=20
@@ -133,10 +128,10 @@ jobs:
       fail-fast: false
       matrix:
         runs-on: [ubuntu-latest, macos-13, windows-latest]
-        pypy-version: ["3.9"]
+        pypy-version: ["3.10"]
         include:
         - runs-on: ubuntu-latest
-          pypy-version: "3.7"
+          pypy-version: "3.8"
 
     steps:
     - uses: actions/checkout@v4

--- a/skbuild/command/build_py.py
+++ b/skbuild/command/build_py.py
@@ -73,7 +73,7 @@ class build_py(set_build_base_mixin, _build_py):
             try:
                 (package_dir, checked) = packages[package]
             except KeyError:
-                package_dir = self.get_package_dir(package)  # type: ignore[no-untyped-call]
+                package_dir = self.get_package_dir(package)
                 checked = False
 
             if not checked:

--- a/skbuild/platform_specifics/windows.py
+++ b/skbuild/platform_specifics/windows.py
@@ -215,7 +215,7 @@ def _get_msvc_compiler_env(vs_version: int, vs_toolset: str | None = None) -> Ca
         return __get_msvc_compiler_env_cache[cache_key]
 
     if hasattr(monkey, "patch_for_msvc_specialized_compiler"):
-        monkey.patch_for_msvc_specialized_compiler()  # type: ignore[no-untyped-call]
+        monkey.patch_for_msvc_specialized_compiler()
 
     vc_dir = find_visual_studio(vs_version)
     vcvarsall = os.path.join(vc_dir, "vcvarsall.bat")

--- a/skbuild/setuptools_wrap.py
+++ b/skbuild/setuptools_wrap.py
@@ -250,7 +250,7 @@ def _parse_setuptools_arguments(
 
     return (
         display_only,
-        dist.help_commands,  # type: ignore[attr-defined]
+        bool(dist.help_commands),
         dist.commands,
         dist.hide_listing,  # type: ignore[attr-defined]
         dist.force_cmake,  # type: ignore[attr-defined]

--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -218,7 +218,7 @@ def execute_setup_py(project_dir, setup_args, disable_languages_test=False):
     to_clear = getattr(
         distutils.dir_util, "SkipRepeatAbsolutePaths", getattr(distutils.dir_util, "_path_created", None)
     )
-    assert to_clear, "Must have one of the two supported clearing mechanisms"
+    assert to_clear is not None, "Must have one of the two supported clearing mechanisms"
     to_clear.clear()
 
     # Clear _PYTHON_HOST_PLATFORM to ensure value sets in skbuild.setuptools_wrap.setup() does not

--- a/tests/__init__.py
+++ b/tests/__init__.py
@@ -215,7 +215,11 @@ def execute_setup_py(project_dir, setup_args, disable_languages_test=False):
     """
 
     # See https://stackoverflow.com/questions/9160227/dir-util-copy-tree-fails-after-shutil-rmtree
-    distutils.dir_util._path_created.clear()  # type: ignore[attr-defined]
+    to_clear = getattr(
+        distutils.dir_util, "SkipRepeatAbsolutePaths", getattr(distutils.dir_util, "_path_created", None)
+    )
+    assert to_clear, "Must have one of the two supported clearing mechanisms"
+    to_clear.clear()
 
     # Clear _PYTHON_HOST_PLATFORM to ensure value sets in skbuild.setuptools_wrap.setup() does not
     # influence other tests.

